Makefile

```diff
@@ -75,6 +75,10 @@ verify: unit-test
 unit-test:
 	go test -v -cover ./pkg/... ./test/utils/credentials
 
+.PHONY: unit-test-v2
```

Comment:
Are we doing this separation only for the unit tests? Or is there a plan to pass this optional parameter so that the build and make container commands also use the v2 implementation instead of v1?

Reply:
should combine with original `unit-test`, otherwise we are not running ut on v2 code

Reply:
I think we need to also add this flag to other tests, otherwise it would lose test coverage on all v2 code

Reply:
I agree that the unit-tests should be combined.

For the other tests (e2e, in particular), would we need to build separate container images with v2 in the interim? Or should I just simply add duplicate targets for the v2 images? If the latter, how should we handle the image version, e.g. add a `-v2` suffix to IMAGE_VERSION or use a separate image version like `vNext`? I assume Helm charts should be added to a directory with the same name as IMAGE_VERSION to enable deployment of the `v2` binaries.

Reply:
one test pipeline could not test v1 & v2, need to set up another test pipeline(e.g. pull-azuredisk-csi-driver-e2e-single-az-v2). I think in the beginning, could set up sanity and integration tests against v2: https://github.com/kubernetes-sigs/azuredisk-csi-driver/tree/master/test/sanity

Reply:
I've push changes to enable sanity and integration testing of the v2 driver.

Reply:
can we enable v2 tests after running v1 tests  in sanity and integration testing, seems it's still only running v1 by default now. For e2e tests, need a new test pipeline in test-infra.

Also could you remove the Draft tag, then it will run sanity and integration testing

Reply:
Sanity and integration tests are now run against both v1 & v2 by default.

Reply:
I separated the sanity & integration targets again and instead added separate pipelines for the v2 tests. See the [discussion](https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/727#discussion_r581476359) below.